flink-ai-flow/wiki_doc/Tutorial.md

```diff
@@ -0,0 +1,383 @@
+
+The `dependecies` directory is used to save python/java/go dependencies that will be used by our workflow.
+
+The `resources` directory is for saving all other files(e.g., datasets) that will be used by the project.
+
+The `project.yaml` is for configuring the whole project. 
+
+Now let's go over above config files one by one.
+
+### project.yaml
+Here is an example of the project.yaml for tutorial project.
+```yaml
+project_name: tutorial_project
+server_ip: localhost
+server_port: 50051
+blob:
+  blob_manager_class: ai_flow_plugins.blob_manager_plugins.local_blob_manager.LocalBlobManager
+```
+In `project_name`, we define the project name, which will be the default namespace of workflows in this project as well. 
+
+> Namespace in Flink AI Flow is used for isolation. Each workflow can only send events to its own namespace while it can listen on multiple namespaces. The reason for enabling listening on multiple namespaces is that the workflow could be triggered by external events from notification service.
+
+For `server_ip` and `server_port`,  they tell where the `AIFlowServer` will starts on.
+
+Then, we configure the `blob` property which tells Flink AI Flow which manages the execution environment(i.e., local or remote env) of workflows in this project.
+
+Here we choose to use `LocalBlobManager` and as a result, the project will run locally. 
+
+We currently also provide `OssBlobManager` which allows users to run their workflow remotely.
+
+### tutorial_workflow.yaml 
+Next, we will introduce how to write the workflow configuration yaml file.
+```yaml
+train:
+  job_type: python
+  periodic_config:
+    interval: '0,0,0,60' # The train job will be executed 60s
+
+validate:
+  job_type: python
+
+predict:
+  job_type: flink
+  properties:
+    run_mode: cluster
+    flink_run_args: #The flink run command args(-pym, -pyexec etc.). It's type is List.
+      - -pyexec
+      - /path/to/bin/python # path to your python3.7 executable path
+```
+In the workflow yaml, we define the properties of each job. 
+
+For `train` job, its job type is `python`, which means the user defined logic in this job will be executed using `python`. Besides, we set the `periodic_config` to be `interval: '0,0,0,60'` which means the job will be executed every 60s.
+
+For `validate` job, we only config its job type to be `python`.
+
+For `predict` job, we set its job type to be `flink`, which means this job is a flink job. In addition, to use flink, we need to set some flink-specific properties including `run_mode` (cluster or local) and `flink_run_args`. For the `flink_run_args`, we follow the list format of `yaml` to add the args such as the `-pyexec`.
+
+## Writing the Workflow Code
+
+Now, let's start coding to implement workflow described in Target section
+
+### Importing Modules
+A workflow Flink AI Flow is just a Python script. Let’s create `tutorial_workflow.py` and start by importing the libraries we will need.
```

Comment:
A workflow in Flink AI Flow

Reply:
In Flink AI Flow, a workflow is just a Python script.